controller/network/network.go

```diff
@@ -526,18 +526,23 @@ func (network *Network) UpdateCircuit(circuit *Circuit) (*Circuit, error) {
 		IngressId: circuit.IngressId,
 		EgressId:  circuit.EgressId,
 	}
-	network.setLinks(circuit2)
+	if err := network.setLinks(circuit2); err != nil {
+		return nil, err
+	}
 	return circuit2, nil
 }
 
-func (network *Network) setLinks(circuit *Circuit) {
+func (network *Network) setLinks(circuit *Circuit) error {
 	if len(circuit.Path) > 1 {
 		for i := 0; i < len(circuit.Path)-1; i++ {
 			if link, found := network.linkController.leastExpensiveLink(circuit.Path[i], circuit.Path[i+1]); found {
 				circuit.Links = append(circuit.Links, link)
```

Comment:
Return error instead of ignoring it when a link can't be found. If a link isn't found we will panic later as array indices won't line up.